x/stakeibc/keeper/reward_converter.go

```diff
@@ -85,24 +85,28 @@ func (k Keeper) CalculateRewardsSplitBeforeRebate(
 	strideFee := sdk.NewIntFromUint64(k.GetParams(ctx).StrideCommission)
 	strideFeeRate := sdk.NewDecFromInt(strideFee).Quo(sdk.NewDec(100))
 
-	// It shouldn't be possible to have 0 delegations (since there are rewards and there was a community pool stake)
+	// Get supply of stTokens to determine the portion of TVL that the community pool liquid stake makes up
+	stDenom := utils.StAssetDenomFromHostZoneDenom(hostZone.HostDenom)
+	stTokenSupply := k.bankKeeper.GetSupply(ctx, stDenom).Amount
+
+	// It shouldn't be possible to have 0 token supply (since there are rewards and there was a community pool stake)
 	// This will also prevent a division by 0 error
-	if hostZone.TotalDelegations.IsZero() {
+	if stTokenSupply.IsZero() {
 		return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrDivisionByZero,
 			"unable to calculate rebate amount for %s since total delegations are 0", hostZone.ChainId)
 	}
 
 	// It also shouldn't be possible for the liquid stake amount to be greater than the full TVL
-	if rebateInfo.LiquidStakeAmount.GT(hostZone.TotalDelegations) {
+	if rebateInfo.LiquidStakedStTokenAmount.GT(stTokenSupply) {
```

Comment:
Just curious, what happens in this case? Should we add any protection to setting `rebateInfo` to make sure this case doesn't trigger?

Reply:
realistically I don't think it should ever trigger! We could add a check when registering a rebate that the LS amount is less than the current supply, but the supply can change so it doesn't perfectly eliminate this case

Reply:
just added the check during registration

Reply:
That makes sense to me, thanks for adding it! 

Out of curiosity, do you know what would happen if it did trigger? Is there a way to handle it gracefully (e.g. just not pay a rebate)?

Reply:
We could just admin update the rebate struct if it did trigger (and then next epoch the rebate would be ignored)

But imo, we should throw and error here instead of silently failing since it's unexpected!

Reply:
Makes sense to me, marking as resolved!